vault/expiration.go

```diff
@@ -453,6 +446,44 @@ func (c *Core) stopExpiration() error {
 	return nil
 }
 
+func (m *ExpirationManager) leaseView(ctx context.Context, ns *namespace.Namespace) (BarrierView, error) {
+	if ns.ID == namespace.RootNamespaceID {
+		return m.core.systemBarrierView.SubView(expirationSubPath + leaseViewPrefix), nil
+	}
+	return m.core.namespaceMountEntryView(ctx, ns.ID, systemBarrierPrefix+expirationSubPath+leaseViewPrefix)
+}
+
+func (m *ExpirationManager) tokenIndexView(ctx context.Context, ns *namespace.Namespace) (BarrierView, error) {
+	if ns.ID == namespace.RootNamespaceID {
+		return m.core.systemBarrierView.SubView(expirationSubPath + tokenViewPrefix), nil
+	}
+	return m.core.namespaceMountEntryView(ctx, ns.ID, systemBarrierPrefix+expirationSubPath+tokenViewPrefix)
+}
+
+func (m *ExpirationManager) collectLeases() (map[*namespace.Namespace][]string, int, error) {
+	leaseCount := 0
+	existing := make(map[*namespace.Namespace][]string)
+	namespaces, err := m.core.namespaceStore.ListAllNamespaces(m.quitContext, true)
+	if err != nil {
+		return nil, 0, fmt.Errorf("failed to list all namespaces: %w", err)
+	}
+
+	for _, namespace := range namespaces {
+		view, err := m.leaseView(m.quitContext, namespace)
+		if err != nil {
+			return nil, 0, fmt.Errorf("failed to determine lease view for namespace: %w", err)
+		}
+		keys, err := logical.CollectKeys(m.quitContext, view)
```

Comment:
In #1102 I'm adding support for clearing views with pagination. I might hold on that change until Namespaces land given I'm worried it'll conflict, but it'd be great to update this to behave similarly. There's no point collecting all leases from all namespaces in a single batch, better to use pagination and iteratively handle lease revocation. My 2c.